File: 283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums):
        """
        Do not return anything, modify nums in-place instead.
        """
        # print(nums)

        if len(nums) == 1:
            return nums

        for val in nums:
            if val == 0:
                nums.remove(0)
                nums.append(0)
        # print(nums)
        return nums


newMoves = Solution()
print(newMoves.moveZeroes([0, 1, 0, 3, 12]))

File: 283_move-zeroes.c
#include <stdio.h>

void moveZeroes(int *nums, int numsSize)
{
    int nonZero = 0;
    for (int i = 0; i < numsSize; i++)
    {
        if (nums[i] != 0)
        {
            nums[nonZero] = nums[i];
            nonZero++;
        }
    }

    for (int i = nonZero; i < numsSize; i++)
        nums[i] = 0;
}

int main()

{
    int nums[] = {0, 1, 0, 3, 12};
    int numsSize = sizeof(nums) / sizeof(nums[0]);

    moveZeroes(nums, numsSize);

    for (int i = 0; i < numsSize; i++)
        printf("%d ", nums[i]);

    return 0;
}